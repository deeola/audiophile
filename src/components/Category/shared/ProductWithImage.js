import React from 'react'
import ProductButton from './ProductButton'
import { Image, ProductWithImageCont } from './SharedStyling'

const ProductWithImage = ({img, link, np, label, title, subtitle, primary}) => {
  return (
    <ProductWithImageCont data-testid='productimage' primary={primary}>
        <Image src={img} alt={title}>
        </Image>
        <ProductButton  subtitle={subtitle} title={title} label={label} link={link} np={np}  />
    </ProductWithImageCont>
  )
}

export default ProductWithImage

