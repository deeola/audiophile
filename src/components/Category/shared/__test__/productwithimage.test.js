import { render } from "@testing-library/react";
import reactDom from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import ProductWithImage from "../ProductWithImage";

it("renders div correctly", () => {
  const div = document.createElement("div");

  reactDom.render(
    <BrowserRouter>
      <ProductWithImage />
    </BrowserRouter>,
    div
  );
});

it('contains image tag', () => {
    const {getByTestId} = render(
        <BrowserRouter>
        <ProductWithImage data-testid='productimage' />
      </BrowserRouter>
    )

    expect(getByTestId('productimage')).toContainHTML('img')
})

//create snapshot

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ProductWithImage />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
