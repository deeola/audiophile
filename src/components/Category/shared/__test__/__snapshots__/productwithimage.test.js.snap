// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bczRLJ gIwRId"
  data-testid="productimage"
>
  <img
    className="sc-dkzDqf gRVFrI"
  />
  <article
    className="sc-hKMtZM bQLcuJ"
    data-testid="productbutton"
  >
    <div
      className="sc-eCYdqJ bMkDvt"
    >
      <p
        className="sc-iBkjds dIztKO"
        data-testid="np"
      >
         
      </p>
      <h2
        className="sc-jSMfEi imCpJA"
        data-testid="title"
      />
      <p
        className="sc-gKXOVf dgecyH"
        data-testid="subtitle"
      />
    </div>
    <a
      className="sc-ftvSup KzLBC"
      color="primary"
      data-testid="productbutton-undefined"
      hover="primary"
      href=""
      onClick={[Function]}
    />
  </article>
</div>
`;
